From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Manish Bhardwaj <mbhardwaj@nvidia.com>
Date: Fri, 1 Aug 2025 09:51:45 +0000
Subject: [PATCH 100/245] NVIDIA: SAUCE: tegra_hv: add max number of vmid check

Bug 5433066

Change-Id: I98e8444bb422bc3c79784523c03406cdf347ee10
Signed-off-by: Manish Bhardwaj <mbhardwaj@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3427779
Reviewed-by: Suresh Venkatachalam <skathirampat@nvidia.com>
Reviewed-by: svcacv <svcacv@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
(cherry picked from commit 0b43455aba72c45d44ae237980cf92b1f0f9bbbc)

diff --git a/drivers/virt/tegra/tegra_hv.c b/drivers/virt/tegra/tegra_hv.c
index c130a91d96933..19ba8d544acf8 100644
--- a/drivers/virt/tegra/tegra_hv.c
+++ b/drivers/virt/tegra/tegra_hv.c
@@ -174,11 +174,14 @@ EXPORT_SYMBOL(tegra_hv_get_ivc_info);
 int tegra_hv_get_vmid(void)
 {
 	struct tegra_hv_data *hvd = get_hvd();
+	int ret = 0;
 
-	if (IS_ERR(hvd))
-		return -1;
+	if (IS_ERR(hvd) || hvd->guestid >= NGUESTS_MAX)
+		ret = -EINVAL;
 	else
-		return hvd->guestid;
+		ret = hvd->guestid;
+
+	return ret;
 }
 EXPORT_SYMBOL(tegra_hv_get_vmid);
 
@@ -411,6 +414,10 @@ static ssize_t vmid_show(const struct class *class,
 	struct tegra_hv_data *hvd = get_hvd();
 
 	BUG_ON(!hvd);
+
+	if (hvd->guestid >= NGUESTS_MAX)
+		hvd->guestid = -EINVAL;
+
 	return snprintf(buf, PAGE_SIZE, "%d\n", hvd->guestid);
 }
 static CLASS_ATTR_RO(vmid);
@@ -547,6 +554,11 @@ static int tegra_hv_setup(struct tegra_hv_data *hvd)
 		return -ENODEV;
 	}
 
+	if (hvd->guestid >= NGUESTS_MAX) {
+		ERR("Failed to read valid guest id\n");
+		return -EINVAL;
+	}
+
 	hvd->hv_class = class_create("tegra_hv");
 	if (IS_ERR(hvd->hv_class)) {
 		ERR("class_create() failed\n");
-- 
2.43.0

